src/features/wallet/send/SendView.tsx

```diff
@@ -43,39 +123,61 @@ const SendView = ({ scanResult }: { scanResult?: QrScanResult }) => {
 
   const setMaxAmount = () => {
     triggerNavHaptic()
+
+    const balance = account?.balance
+    if (!balance) return
+
+    const maxAmount = balance.minus(fee)
+
+    // TODO option to not return currency ticker in balance
+    setAmount(maxAmount.toString(8).slice(0, -4))
   }
 
   const unlockForm = () => {
     setIsLocked(false)
     triggerNavHaptic()
   }
 
-  const submitTxn = () => {
-    // eslint-disable-next-line no-console
-    console.log('address', address)
-    // eslint-disable-next-line no-console
-    console.log('amount', amount)
+  const constructTxn = async (): Promise<string> => {
+    if (type === 'payment') {
+      return makePaymentTxn(getIntegerAmount(), address, getNonce())
+    }
+
+    if (type === 'dc_burn') {
+      return makeBurnTxn(getIntegerAmount(), address, getNonce(), memo)
+    }
+
+    throw new Error('Unsupported transaction type')
+  }
+
+  const handleSubmit = async () => {
+    const txn = await constructTxn()
+    await submitTransaction(txn)
```

Comment:
Do we need to handle any errors here?

Reply:
do you think we should just do a `Logger.error` for that?

Reply:
I think a Logger error would be good, and also maybe show a dialog that something went wrong at least? Or the app will just be stuck right?

I guess its pretty rare that the construct and submit itself would fail though?

Reply:
I can add a logger and an alert